src/libsync/propagateremotemkdir.cpp

```diff
@@ -34,29 +34,18 @@ PropagateRemoteMkdir::PropagateRemoteMkdir(OwncloudPropagator *propagator, const
     , _uploadEncryptedHelper(nullptr)
     , _parallelism(FullParallelism)
 {
-    const auto rootPath = [=]() {
-        const auto result = propagator->remotePath();
-        if (result.startsWith('/')) {
-            return result.mid(1);
-        } else {
-            return result;
-        }
-    }();
     const auto path = _item->_file;
     const auto slashPosition = path.lastIndexOf('/');
     const auto parentPath = slashPosition >= 0 ? path.left(slashPosition) : QString();
 
     SyncJournalFileRecord parentRec;
     bool ok = propagator->_journal->getFileRecord(parentPath, &parentRec);
     if (!ok) {
+        // TODO: not sure how this can be possible, maybe this job should not get created if the parent folder is missing?
```

Comment:
This is more a problem of DB failure. We got a stress test for such cases (testdatabaseerror.cpp). This pushes us to handle this kind of cases as gracefully as possible, in this case that means returning early and maybe have the sync fail later.